protocols/relay/src/behaviour.rs

```diff
@@ -280,6 +282,10 @@ impl Behaviour {
             peer.get_mut().remove(&connection_id);
             if peer.get().is_empty() {
                 peer.remove();
+                self.queued_actions
+                    .push_back(ToSwarm::GenerateEvent(Event::ReservationClosed {
+                        src_peer_id: peer_id,
+                    }));
             }
```

Comment:
```suggestion
            if peer.get_mut().remove(&connection_id) {
                self.queued_actions
                    .push_back(ToSwarm::GenerateEvent(Event::ReservationClosed {
                        src_peer_id: peer_id,
                    }));
            }
            if peer.get().is_empty() {
                peer.remove();
            }
```

We may have multiple connections to a peer, so we should emit the event when a connection closes for which we have an active reservation. 

Maybe it also makes sense to include the connection id in the event?

Reply:
I agree. Additionally, we should also include the connection id of which we had the reservation since there may be multiple reservation to a peer with different connections.

Reply:
To make it complete, the other events should also have connection IDs: `ReservationReqAccepted`, `ReservationReqDenied` and `ReservationTimedOut`.

Reply:
How about we keep the `ReservationClosed` variant without the connection-id in this PR, and do a separate one that adds it for all of the mentioned variants (and discuss there if we want to in the first place)? That way it's consistent.

Reply:
> We may have multiple connections to a peer, so we should emit the event when a connection closes for which we have an active reservation.
Thanks. This makes sense! I'll update the PR.

And I'll create a followup PR to add connection IDs to the events.